engines/director/lingo/lingo-the.cpp

```diff
@@ -1601,6 +1601,11 @@ void Lingo::setTheSprite(Datum &id1, int field, Datum &d) {
 			sprite->_moveable = false;
 		}
 		break;
+	case kTheRect:
+		score->renderSprites(score->getCurrentFrame(), kRenderForceUpdate);
+		channel->_currentPoint = Common::Point(d.u.farr->arr[0].u.i, d.u.farr->arr[1].u.i);
```

Comment:
I do not understand this logic. What is the Lingo for this? Is it "rect of sprite"? If so, then this belongs to D5, and then why are you altering `channel` here and why only with a point?

Reply:
yes, it is rect of sprite. In a D4 movie. I altered channel's point here as that alone introduces an offset in the channel's - hence the sprite's - rect

Reply:
I am confused. What does happen with the other two array elements of a RECT? Also, where is the type checking? What if `d` is not an array?